src/Microsoft.Framework.Runtime/FileGlobbing/ProjectFilesCollection.cs

```diff
@@ -3,121 +3,101 @@
 
 using System.Collections.Generic;
 using System.Linq;
-using Microsoft.Framework.FileSystemGlobbing;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Framework.Runtime.FileGlobbing
 {
     public class ProjectFilesCollection
     {
-        public static readonly string[] DefaultSourcePatterns = new[] { @"**\*.cs" };
-        public static readonly string[] DefaultExcludePatterns = new[] { @"obj\**\*.*", @"bin\**\*.*" };
-        public static readonly string[] DefaultBundleExcludePatterns = new[] { @"obj\**\*.*", @"bin\**\*.*", @"**\.*\**" };
-        public static readonly string[] DefaultPreprocessPatterns = new[] { @"compiler\preprocess\**\*.cs" };
-        public static readonly string[] DefaultSharedPatterns = new[] { @"compiler\shared\**\*.cs" };
-        public static readonly string[] DefaultResourcesPatterns = new[] { @"compiler\resources\**\*" };
-        public static readonly string[] DefaultContentsPatterns = new[] { @"**\*" };
-
-        private static readonly string PropertyNameCode = "code";
-        private static readonly string PropertyNameExclude = "exclude";
-        private static readonly string PropertyNameBundleExclude = "bundleExclude";
-        private static readonly string PropertyNamePreprocess = "preprocess";
-        private static readonly string PropertyNameShared = "shared";
-        private static readonly string PropertyNameResources = "resources";
-        private static readonly string PropertyNameContent = "files";
-
-        private readonly Matcher _sourcesMatcher;
-        private readonly Matcher _preprocessSourcesMatcher;
-        private readonly Matcher _sharedSourceMatcher;
-        private readonly Matcher _resourcesMatcher;
-        private readonly Matcher _contentFilesMatcher;
-        private readonly Matcher _bundleExcludeFilesMatcher;
+        public static readonly string[] DefaultCompilePatterns = new[] { @"**/*.cs" };
+        public static readonly string[] DefaultBundleExcludePatterns = new[] { @"obj/**/*.*", @"bin/**/*.*", @"**/.*/**" };
+        public static readonly string[] DefaultPreprocessPatterns = new[] { @"compiler/preprocess/**/*.cs" };
+        public static readonly string[] DefaultSharedPatterns = new[] { @"compiler/shared/**/*.cs" };
+        public static readonly string[] DefaultResourcesPatterns = new[] { @"compiler/resources/**/*" };
+        public static readonly string[] DefaultContentsPatterns = new[] { @"**/*" };
+        public static readonly string[] DefaultBuiltInExcludePatterns = new[] { "bin/**", "obj/**", "*.kproj" };
```

Comment:
I think we need builtins for compile as well. Possible for most of it /cc @lodejard